compiler/rustc_trait_selection/src/traits/engine.rs

```diff
@@ -129,6 +129,19 @@ impl<'a, 'tcx> ObligationCtxt<'a, 'tcx> {
             .map(|infer_ok| self.register_infer_ok_obligations(infer_ok))
     }
 
+    pub fn eq_no_opaques<T: ToTrace<'tcx>>(
```

Comment:
It's detrimental to use `eq(DefineOpaqueTypes::Yes, ..)` during method probe because it breaks some TAIT tests (and bootstrap!)

Could fix those in the compiler I guess and change the callsites back to `eq` if we're okay w/ it though